src/main/java/com/commercetools/pspadapter/tenant/TenantConfigFactory.java

```diff
@@ -0,0 +1,42 @@
+package com.commercetools.pspadapter.tenant;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Component
+public class TenantConfigFactory {
+
+    private final Environment env;
+
+    @Autowired
+    public TenantConfigFactory(@Nonnull Environment env) {
+        this.env = env;
+    }
+
+    public Optional<TenantConfig> getTenantConfig(@Nonnull String tenantName) {
+        String projectKey = env.getProperty(tenantName + ".ctp.client.projectKey");
+        String clientId = env.getProperty(tenantName + ".ctp.client.clientId");
+        String clientSecret = env.getProperty(tenantName + ".ctp.client.clientSecret");
+        String ppPClientId = env.getProperty(tenantName + ".paypalPlus.client.clientId");
+        String ppPClientSecret = env.getProperty(tenantName + ".paypalPlus.client.clientSecret");
+        String ppPClientMode = env.getProperty(tenantName + ".paypalPlus.client.mode");
+
+        if (isAnyPropertyNull(projectKey, clientId, clientSecret,
+                ppPClientId, ppPClientSecret, ppPClientMode)) {
+            return Optional.empty();
```

Comment:
we have a problem here (and not only): so far we silently create empty objects, so in the future we will have a lot of troubles debugging the issues why the payments are note processed!

We have couple of choices:
  - Add at least logging of _warn_ or _error_ if tenant specific handlers can't be created with explicit reasons why
  - add the error message to response body, like "tenant %s can't be created because of missing parameters XXX"
  - make a custom exception thrown if some tenant configs or tenant factories, or tenant handlers can't be created

Basically, we could aggregate all this solutions to some common solution, but we should not stick with silence.